### ISSUE TITLE
metrics.do_not_track() blocks also metrics.summary()/…

### ISSUE THREAD

**Issue by hynek** [rycus86/prometheus_flask_exporter#42]
I have a blueprint endpoint with an variable rule:

```python
bp = Blueprint("rdap", __name__)

@bp.route("/domain/<string:domain>")
def query(domain: str) -> Tuple[Dict, int]:
    …
```

Since that gives me an overwhelming cardinality, I wanted to drop the domain from the metric. There doesn't seem to be a good way, I wanted to try a custom metric and took this straight from the examples:

```python
@bp.route("/domain/<string:domain>")
@metrics.do_not_track()
@metrics.summary('requests_by_status', 'Request latencies by status',
                 labels={'status': lambda r: r.status_code})
def query(domain: str) -> Tuple[Dict, int]:
    …
```

Unfortunately, that doesn't work: the metric gets created, but not computed.

If I remove the `do_not_track(()`, it starts computing.  This looks like a bug to me since `do_not_track()` should only disable default metrics.  Or am I missing something?  Is this because I'm using blueprints and/or delayed app creation?

**Comment by rycus86** [comment]
Hi! `do_not_track()` disables tracking the endpoint completely (if I remember correctly), and I don't think there's a way to disable the built-in ones on a single endpoint at the moment.

It shouldn't be too hard to add this though, I'll try to give it a go this week.
I see now that the readme has examples with both `do_not_track` and a new metric, I think that was a mistake.
An alternative is to set `export_defaults=False` when you create the `PrometheusMetrics` instance and register default metrics yourself with `register_default` .

**Comment by hynek** [comment]
Thanks for getting back to me so fast!

Yeah, I guess it's a docs bug then. Especially because it literally says:

> only the counter is collected, not the default metrics

:)

That behavior would also make more sense I think.

**Comment by rycus86** [comment]
OK, I think I actually broke this on the last release with https://github.com/rycus86/prometheus_flask_exporter/commit/6b5424c5f37ecf36de6612f0283a00147625f4ac
Now I reverted this logic, so it does what it says in the README. If needed both the default and custom metrics can be suppressed with the new `@metrics.exclude_all_metrics()` decorator added in  https://github.com/rycus86/prometheus_flask_exporter/commit/e180b11a96bba59a46c9d8fff33d620085c0004b which will be available in version `0.12.0` in a few minutes.

**Comment by hynek** [comment]
Thanks!
